src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -374,62 +374,45 @@ SafeToPushdownUnionSubquery(PlannerRestrictionContext *plannerRestrictionContext
 
 
 /*
- * FindUnionAllVar finds the variable used in union all for the side that has
- * relationRteIndex as its index and the same varattno as the partition key of
- * the given relation with relationOid.
+ * RangeTableOffsetCompat returns the range table offset(in glob->finalrtable) for the appendRelInfo.
+ * For PG < 13 this is a no op.
  */
-static Var *
-FindUnionAllVar(PlannerInfo *root, List *appendRelList, Oid relationOid,
-				Index relationRteIndex, Index *partitionKeyIndex)
+int
+RangeTableOffsetCompat(PlannerInfo *root, AppendRelInfo *appendRelInfo)
```

Comment:
The following query can be pushed down on PG 12 but not on PG 13. Could it be related to this function where we check only the first inherited entry?
```SQL

     SELECT *
FROM
  (SELECT user_id  AS user_id
   FROM
     (SELECT user_id  AS user_id
      FROM users_table
      UNION ALL SELECT user_id AS user_id
      FROM users_table) AS bar
   UNION ALL SELECT user_id  AS user_id
   FROM users_table) AS fool LIMIT 1;
```